Assets/UnityIO.cs

```diff
@@ -0,0 +1,48 @@
+using UnityEngine;
+using System;
+using System.IO;
+using System.Collections;
+
+public class UnityIO
+{
+	public class Response {
+		public string error;
+		public byte[] data;
+
+		public Response(string e, byte[] d) {
+			error = e;
+			data = d;
+		}
+		public Response(string e, string d) {
+			error = e;
+			data = Convert.FromBase64String(d);
+		}
+	}
+
+	public delegate void IORequestCallback(object userData, Response response);
+
+	public IEnumerator FetchData(Uri uri, IORequestCallback callback, object userData) {
+
+		bool networkRequest = false;
+		Response response;
+		if (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps) {
+			networkRequest = true;
+		}
+
+		if (networkRequest) {
+			WWW request = new WWW(uri.AbsoluteUri);
```

Comment:
Judging by the docs, `UnityWebRequest` has more granular access to parameters that we will probably be interested in, like request timeout and response codes. It seems to me that we will probably want to use `UnityWebRequest` over `WWW` in the long term.